src/file-events/headers/generic_fsnotifier.h

```diff
@@ -178,13 +175,13 @@ class TerminateCommand : public Command {
     }
 };
 
-struct JniConstants {
-    JniConstants(JNIEnv* env);
-    void unload(JNIEnv* env);
+class JniConstants : public JniSupport {
```

Comment:
I think we should move `JniConstants` to `jni_support` as well.

Reply:
I think we should rename it to `GenericClasses` or something. Or maybe move them to `AbstractServer` instead? Now it's pretty simple to store a reference to a class that you intend to use in a given context.

Reply:
Storing is simple, though there is only one `JNI_OnLoad` method. The method currently lives in `generic_fsnotifier.cpp`, maybe that method should go somewhere else to initialize the logging and the jni_constants?

I wouldn't move them to `AbstractServer` since I think it is better to store the classes references on load.

Reply:
Alright, I moved them to `services.cpp`.